core/app/models/mno_enterprise/tenant_config.rb

```diff
@@ -410,6 +74,19 @@ def self.reconfigure_mnoe!
       ActionMailer::Base.smtp_settings = Settings.system.smtp.to_hash
     end
 
+    # Update the JSON schema with values available after initialization
+    # TODO: Refactor to use Proc in the JSON Schema and call them
+    def self.refresh_json_schema!
+      # Re-evaluate the availables locales as the list is only populated after initialization
+      available_locales = I18n.available_locales.map(&:to_s).select {|x| x =~ /[[:alpha:]]{2}-[A-Z]{2}/}
+      locale_map = Hash[available_locales.map {|l| [l, I18n.t('language', locale: l, fallback: false, default: nil) || l] }]
+
+      json_schema['properties']['system']['properties']['i18n']['properties']['available_locales']['items']['enum'] = available_locales
```

Comment:
I would suggest ?
```
props = json_schema['properties']['system']['properties']['i18n']['properties']
props['available_locales']['items']['enum'] = available_locales
props['preferred_locale']['enum'] = available_locales
props['available_locales']['x-schema-form']['titleMap'] = locale_map
props['preferred_locale']['x-schema-form']['titleMap'] = locale_map
```

Reply:
@x4d3 note that this section of the code will be refactored anyway. We will use Procs in the schema values to make it dynamic. So we should be able to remove this portion of the code eventually.

Reply:
done